tests/serial/map-if-necessary/three-solvers/mixed-substeps/ParallelImplicit.cpp

```diff
@@ -0,0 +1,34 @@
+#ifndef PRECICE_NO_MPI
+
+#include "../helper.hpp"
+#include "testing/Testing.hpp"
+
+#include <precice/precice.hpp>
+#include <vector>
+
+BOOST_AUTO_TEST_SUITE(Integration)
+BOOST_AUTO_TEST_SUITE(Serial)
+BOOST_AUTO_TEST_SUITE(MapIfNecessary)
+BOOST_AUTO_TEST_SUITE(ThreeSolvers)
+BOOST_AUTO_TEST_SUITE(MixedSubsteps)
+
+BOOST_AUTO_TEST_CASE(ParallelImplicit)
+{
+  PRECICE_TEST("A"_on(1_rank), "B"_on(1_rank), "C"_on(1_rank));
+
+  std::vector<int> readMappings{3, 0, 3, 0, 3, 0};
```

Comment:
```suggestion
  std::vector<int> readMappings{3, 1, 3, 1, 3, 1};
```
The new data from B still needs to be mapped??